src/features/farming/trader/Trader.tsx

```diff
@@ -0,0 +1,162 @@
+
+  const handleOpenModal = () => {
+    if (hasAlreadyTraded(state)) {
+      setModalState("alreadyTraded");
+      return;
+    }
+
+    setModalState("intro");
+  };
+
+  const handleTrade = () => {
+    const offer = state.tradeOffer;
+
+    if (offer) {
+      offer.ingredients?.map((ingredient) => {
+        const item = ITEM_DETAILS[ingredient.name];
+        setToast({
+          icon: item.image,
+          content: `-${ingredient.amount}`,
+        });
+      });
+
+      setToast({
+        icon: ITEM_DETAILS[offer.name].image,
+        content: `+1`,
+      });
+    }
+
+    gameService.send("item.traded");
+    setModalState("tradeCompleted");
+  };
+
+  const handleCloseModal = () => {
+    setModalState("closed");
+  };
+
+  const ModalContent = () => {
+    if (modalState === "tradeCompleted") {
+      return (
+        <Content title="Thanks!">
+          <p className="sm:text-sm p-2">
+            It was a pleasure doing business with you. I will see you again
+            soon!
+          </p>
+        </Content>
+      );
+    }
+
+    if (modalState === "alreadyTraded") {
+      return (
+        <Content title="We've already traded!">
+          <p className="sm:text-sm p-2">
+            {`I have nothing left to trade. I'm just here catching up with old
+            friends!`}
+          </p>
+        </Content>
+      );
+    }
+
+    if (modalState === "showOffer") {
+      return (
+        <Offer
+          inventory={state.inventory}
+          offer={state.tradeOffer as TradeOffer}
+          onCraft={handleTrade}
+        />
+      );
+    }
+
+    if (modalState === "intro") {
+      const endDateLocale = new Date(
+        state.tradeOffer?.endAt as string
+      ).toLocaleDateString();
+
+      return (
+        <Content title="Greetings friend!">
+          <p className="sm:text-sm p-2">
+            I am a nomad trader. I travel all over these lands collecting items
+            to trade.
+          </p>
+          <p className="sm:text-sm p-2">
+            What I have to offer you today will only be available until{" "}
+            {endDateLocale}.
+          </p>
+          <Button onClick={() => setModalState("showOffer")}>
+            {`Let's trade!`}
+          </Button>
+        </Content>
+      );
+    }
+  };
+
+  return (
+    <div
+      className="z-100 absolute"
+      id="trader"
+      style={{
+        width: `${GRID_WIDTH_PX * 1}px`,
+        left: `${GRID_WIDTH_PX * 1}px`,
+        top: `${GRID_WIDTH_PX * 4.5}px`,
+      }}
+    >
+      <div className="cursor-pointer hover:img-highlight z-10">
+        <img
+          src={traderImage}
+          alt="trader"
+          onClick={handleOpenModal}
+          className="w-full"
+        />
+      </div>
+
+      <Modal centered show={modalState !== "closed"} onHide={handleCloseModal}>
+        <Panel>
+          <img
+            src={close}
+            className="h-6 top-4 right-4 absolute cursor-pointer"
```

Comment:
Consider adding a z-index `z-10`, I can't seem to click this on the trade modal

<img width="331" alt="Screen Shot 2022-06-21 at 10 30 35 am" src="https://user-images.githubusercontent.com/11745561/174692712-3e848d68-5d8f-4adf-8105-5af114dc528b.png">